### PR TITLE
This MR can make it work in vs2015

### PR DESCRIPTION
**Pull request by ChenXiaoTemp**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,148 @@
+# CMAKE generated file: DO NOT EDIT!
```
Comment (ppwwyyxx) [inline Makefile]: Please do not include auto-generated file

File: README.md [hunk 2]
```diff
@@ -19,10 +20,25 @@ which is also the one used by [AutoStitch](http://matthewalunbrown.com/autostitc
 Eigen, CImg and FLANN are header-only, to simplify the compilation on different platforms.
 CImg and libjpeg are only used to read and write images, so you can easily get rid of them.
 
+### Steps
+
+* Download Cmake and install the cmake
+* Download Eigen and unzip it in panorama/Eigen, please ensure that panorama/Eigen/Eigen/Dense exists.
```
Comment (ppwwyyxx) [inline README.md]: Definitely we should allow users to use Eigen in the system. 
So users should set eigen path by environment vars only if they want to download Eigen by themselves.

File: README.md [hunk 3]
```diff
@@ -19,10 +20,25 @@ which is also the one used by [AutoStitch](http://matthewalunbrown.com/autostitc
...
+
 ### Compile:
 Linux / OSX (gcc or clang):
 ```
-$ cd src; make
+$ cmake .
+$ make
```
Comment (ppwwyyxx) [inline README.md]: Could you please not override the original Makefile and keep the original make-based build system?

File: src/MSVC/vs.h [hunk 4]
```diff
@@ -0,0 +1 @@
+#define not !
```
Comment (ppwwyyxx) [inline src/MSVC/vs.h]: Adding two separate source directories for these several lines seems too much. Could you just merge this file into common.hh?

File: src/lib/imgio.cc [hunk 5]
```diff
@@ -4,7 +4,6 @@
 #include <cstdlib>
 #include <vector>
 #define cimg_display 0
-#define cimg_use_jpeg
```
Comment (ppwwyyxx) [inline src/lib/imgio.cc]: Most users would definitely need jpeg support. I think we should keep it enabled by default.

File: src/stitch/blender.cc [hunk 6]
```diff
@@ -42,7 +42,7 @@ Mat32f LinearBlender::run() {
 		memset(weight.ptr(), 0, target_size.y * target_size.x * sizeof(float));
 		fill(target, Color::BLACK);
 #pragma omp parallel for schedule(dynamic)
-		REP(k, images.size()) {
+		REP(k, (int)images.size()) {
```
Comment (ppwwyyxx) [inline src/stitch/blender.cc]: Why is this necessary? Openmp on windows does not support `size_t`?

File: README.md (hunk 3 above)
Comment (ChenXiaoTemp) [inline README.md]: I am not sure ,whether the original Makefile is generated by cmake, because I found you have provide a CMakeList.txt , if it is generated by cmake, can we override it ?

File: src/lib/imgio.cc (hunk 5 above)
Comment (ChenXiaoTemp) [inline src/lib/imgio.cc]: OK, I will fix it.

File: src/stitch/blender.cc (hunk 6 above)
Comment (ChenXiaoTemp) [inline src/stitch/blender.cc]: Not the problem of size_t, but that the iteration related to  Openmp on windows need int.

File: README.md (hunk 2 above)
Comment (ChenXiaoTemp) [inline README.md]: OK , I will fix it.I need some time.

File: Makefile (hunk 1 above)
Comment (ChenXiaoTemp) [inline Makefile]: OK, I will fix it.

File: README.md (hunk 3 above)
Comment (ppwwyyxx) [inline README.md]: The original Makefile is not auto-generated. It is just meant to be an alternative way to build the project.

File: src/lib/imgio.cc (hunk 5 above)
Comment (ppwwyyxx) [inline src/lib/imgio.cc]: If it's not easy to support, you can use macro to disable jpeg only on windows.

**Comment by ChenXiaoTemp** [comment]
@ppwwyyxx Could you please continue review it？ I have fixed the issues.

**Comment by ppwwyyxx** [comment]
`src/CMakeLists.txt` is having a huge conflict with master. More specifically, does any of the `find_package` function in cmake work for windows? If they don't work for windows I can handle the conflicts later by putting them into `if`.

**Comment by ppwwyyxx** [comment]
Currently the cmake build is failing on linux because:
1. `find_package` is not used to find Eigen3, and cmake doesn't respect my `EIGEN3_INCLUDE_DIR` environment variables. It looks like you should use `if(DEFINED ENV{EIGEN3_INCLUDE_DIR})` in CMakeLists.txt instead.
2. jpeg is not linked correctly. In `src/CMakeLists.txt` jpeg should get enabled when not on windows.

Do you have any linux machine for testing? If not I can try fixing them.

**Comment by ChenXiaoTemp** [comment]
@ppwwyyxx Please continue review. I have fixed the problem. And I test cmake in linux. Could you test it again, if any problem , please tell me.

**Comment by ppwwyyxx** [comment]
So I see `find_package(JPEG REQUIRED)` is kept in `src/CMakeLists.txt`. Does it work on windows or not? If yes, we don't have to disable jpeg in source code.

**Comment by ChenXiaoTemp** [comment]
In the previous version, I do not use JPEG , it works for windows. I will test it tonight when I go home. Please wait I finished test it.

**Comment by ChenXiaoTemp** [comment]
@ppwwyyxx  Sorry, currently, the jpeg cannot work in windows. So I use the NOT_USE_JPEG_IN_WINDOWS to ignore jpeg in windows. COuld you check it?

**Comment by ChenXiaoTemp** [comment]
Without the jpeg, others works well in vs 2015.
